HouseRobber.java

```diff
@@ -1,3 +1,17 @@
+
+// Time Complexity : O(n), n = number of houses
+// Space Complexity : O(n)
```

Comment:
Are we using any auxiliary DS?

Reply:
Yes, dp[][] is auxiliary space, I think. No?

Reply:
Correct. Try solving it in O(1).

Reply:
O(1) solution is is the same file below it. Kindly check and confirm.